mainwindow.cpp

```diff
@@ -54,32 +54,18 @@ MainWindow::MainWindow(QSettings &settings)
         item->setSizeHint(QSize(m_ui->listWidgetTabs->width(), m_ui->listWidgetTabs->sizeHintForRow(0)));
 
     m_ui->attackModeTabWidget->setCurrentWidget(m_ui->defaultModeTab);
+    m_ui->attackModeTabWidget->tabBar()->installEventFilter(this);
 
-    /*
-    // We add a button to the toolbar but this button is not simple. It has
-    // menu. And that menu drops like from menu button. Just QAction could not
-    // do it. Just QPushButton looses it's main action for popping
-    // menu. Solution with the best looking is QToolButton. QtCreator could not
-    // put such element on tool bar. It is possible to put it by hands through
-    // .ui file but it does not seem to be reliable. So we make it here in
-    // code.
-    //
-    // We create a desired menu.
-    QMenu *sessionMenu = new QMenu(this);
-    sessionMenu->addAction(m_ui->actionNew_Session);
-    sessionMenu->addAction(m_ui->actionSave_Session);
-    // We create a button.
+    // Multiple sessions management menu
+    m_sessionMenu = new Menu(this);
     QToolButton *sessionMenuButton = new QToolButton(this);
```

Comment:
You don't delete either of these as far as I can see. How about making them static member vars instead of pointers so you don't have to worry about them?